### PR TITLE
fix(cmd/main.py): use correct return value

### PR DESCRIPTION
**Pull request by sshedi**
## Proposed Commit Message
```
fix(cmd/main.py): use correct return value
```

## Additional Context
```
$ cloud-init -d modules -m init
Above command returns non zero exit status
```

## Test Steps
```
root@ph4dev [ ~/cloud-init ]# cloud-init -d modules -m init
2024-02-22 19:10:07,659 - handlers.py[DEBUG]: start: modules-init: running modules for init
2024-02-22 19:10:07,659 - util.py[DEBUG]: Reading from /proc/uptime (quiet=False)
...
Cloud-init v. 23.4 running 'modules:init' at Thu, 22 Feb 2024 19:10:07 +0000. Up 26599.38 seconds.
[]

root@ph4dev [ ~/cloud-init ]# echo $?
1
```

## Checklist

- [x] My code follows the process laid out in [the documentation](https://cloudinit.readthedocs.io/en/latest/development/index.html)
- [] I have updated or added any [unit tests](https://cloudinit.readthedocs.io/en/latest/development/testing.html) accordingly
- [x] I have updated or added any [documentation](https://cloudinit.readthedocs.io/en/latest/development/contribute_docs.html) accordingly

## Merge type

- [x] Squash merge using "Proposed Commit Message"
- [ ] Rebase and merge unique commits. Requires commit messages per-commit each referencing the pull request number (#<PR_NUM>)

### CODE REVIEW

**Comment by sshedi** [comment]
I'm not sure how to add a test for this. Any inputs on the same would be helpful. Thanks.

File: cloudinit/cmd/main.py [hunk 1]
```diff
@@ -786,7 +786,9 @@ def status_wrapper(name, args, data_d=None, link_d=None):
         else:
             errors = ret
 
-        v1[mode]["errors"] = [str(e) for e in errors]
+        v1[mode]["errors"] = []
+        for e in errors:
+            v1[mode]["errors"].append(str(e))
```
Comment (a-dubs) [inline cloudinit/cmd/main.py:791]: What is the purpose of this change? The new code is functionally identical to the code it replaces.

**Comment by a-dubs** [comment]
Also, could you please modify your proposed commit message section so that your commit message appears as a code block like the following:
```
fix(cmd/main.py): use correct return value
```

And could you move the 
```
$ cloud-init -d modules -m init
Above command returns non zero exit status
```
information in the PR description to the `Additional Context` section and add some more context for why this change is needed and what the change is seeking to accomplish?

Thanks!

File: cloudinit/cmd/main.py (hunk 1 above)
Comment (sshedi) [reply cloudinit/cmd/main.py:791]: `errors` can be an integer value here, not always a list.

**Comment by sshedi** [comment]
> @sshedi Do you use this for something? This doesn't normally run during boot, so I'm curious what you use it for.

In one of our CI/CD pipelines, I tried adding:
```
echo -e "\nRun cloud-init with all configured modules"
for m in init config final; do
  cmd="cloud-init --file user-data -d modules --mode $m"
  if $cmd; then
    echo -e "\nPASS: $cmd"
  else
    echo -e "\nFAIL: $cmd" 1>&2
  fi
done
```
And caught this issue. Nothing major, but I think the return value should be appropriate.

**Comment by sshedi** [comment]
> Also, could you please modify your proposed commit message section so that your commit message appears as a code block like the following:
> 
> ```
> fix(cmd/main.py): use correct return value
> ```
> 
> And could you move the
> 
> ```
> $ cloud-init -d modules -m init
> Above command returns non zero exit status
> ```
> 
> information in the PR description to the `Additional Context` section and add some more context for why this change is needed and what the change is seeking to accomplish?
> 
> Thanks!

Thanks for the input and suggestions. Previously while raising PRs, this section was simple. I feel it's a bit complex now.
I understand the intent behind this but writing good commit message is already a hard task and filling this template is a bit confusing.

I will try.

File: cloudinit/cmd/main.py (hunk 1 above)
Comment (sshedi) [reply cloudinit/cmd/main.py]: Indeed there was in issue in the code change, fixed it. Thanks.

**Comment by aciba90** [comment]
Thanks, @sshedi, for raising this issue and proposing a fix. I am closing this one in favor of #5017, which includes a correct fix and integration tests. I hope that's okay from your side.

**Comment by sshedi** [comment]
> Thanks, @sshedi, for raising this issue and proposing a fix. I am closing this one in favor of #5017, which includes a correct fix and integration tests. I hope that's okay from your side.

No problem. Thanks for fixing it.
